### PR TITLE
[TG-9400][UFC] Add an assume-no-exceptions-thrown option

### PR DESCRIPTION
**Pull request by romainbrenguier**
This adds an option to JBMC to replace throw instructions by `assert(false);assume(false);`
so that execution branches are not analyzed further than the first `throw`. 
This is mostly useful for performances, but the full benefit of this will only show once   https://github.com/diffblue/cbmc/pull/5097 is in.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [r] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [na] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Comment by smowton** [comment]
I note this doesn't quite do what it says: it actually *asserts* that no exception is thrown, rather than assuming it (TG of course also uses --cover, translating that into an assumption)

**Comment by romainbrenguier** [comment]
@smowton 
> I note this doesn't quite do what it says: it actually asserts that no exception is thrown, rather than assuming it (TG of course also uses --cover, translating that into an assumption)

It actually does both if I'm not mistaken. Should we rename the option to `--assert-no-exceptions-thrown`?

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: e5ba652e3e667a9223c1d000441fc916c8fe2377).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/128707994
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

**Comment by smowton** [comment]
Yes I think that's better (the assert/assume pattern matches what the Java frontend does for null-pointer derefs for example)

**Review by peterschrammel — APPROVED** [review]
lgtm

File: jbmc/src/java_bytecode/java_bytecode_language.h [hunk 1]
```diff
@@ -150,6 +154,56 @@ class lazy_class_to_declared_symbols_mapt
   std::unordered_multimap<irep_idt, symbolt> map;
 };
 
+struct java_bytecode_language_optionst
+{
+  explicit java_bytecode_language_optionst(
```
Comment (peterschrammel) [inline jbmc/src/java_bytecode/java_bytecode_language.h]: explicit not required

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 2d4842f5259db71cf49339fe3caeb0362a0fcd27).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/128776632
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: 6990f9bffe9f53d8847a8174f4801094b7865d64).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/129047220
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

**Review by smowton — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: b433f5c688e0c5ce647448dd72f1d5e534db3c08).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/129121599
Status will be re-evaluated on next push.
Common spurious failures include: the cbmc commit has disappeared in the mean time (e.g. in a force-push); the author is not in the list of contributors (e.g. first-time contributors); compatibility was already broken by an earlier merge.

**Review by segun3d — COMMENTED** [review]
QA is satisfied with this PR

**Pull request merged**